tests/test_generics.py

```diff
@@ -1508,6 +1508,22 @@ class MyInt(int):
     ReferenceModel[MyInt]
 
 
+def test_generic_with_referenced_generic_union_type_bound():
+    T = TypeVar('T', bound=Union[str, int])
```

Comment:
From `typing` docs there is an example of a union bound, so I think it is completely correct to test this case 👍